contracts/src/v0.7/dev/KeeperRegistryDev.sol

```diff
@@ -98,9 +99,11 @@ contract KeeperRegistryDev is
     uint32 checkGasLimit;
     uint24 stalenessSeconds;
     uint16 gasCeilingMultiplier;
+    bool keepersMustTakeTurns;
   }
 
   struct PerformParams {
+    Config config;
```

Comment:
Have you tried only adding the `keepersMustTakeTurns` struct to PerformParams instead of the entire Config struct? I can see we will save reading the config from storage in the `calculatePayment` method, but post-EIP-2929 that only really saves us 100 gas. Compared to the extra memory we have to allocate and pay for to keep the config in memory. Not sure if that is enough to save on gas, I'd jut profile it in a test.

Reply:
Did some profiling [here](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/test/v0.7/KeeperRegistry.test.ts#L656).

- On develop (No keeperMustTakeTurns): 77130

- Storing whole config in PerformParams, pass to calculatePayment: 77107 (-23)

- calculatePayment reads config from storage, performParams includes keeperMustTakeTurns: 77408 (+278)

- performParams unchanged, calculatePayment reads config from storage, prePerformUpkeep also reads 'keeperMustTakeTurns' from storage: 77342 (+212)

So the current approach seems like the most gas efficient according to hardhat